browser/speedreader/speedreader_tab_helper.cc

```diff
@@ -258,6 +260,50 @@ void SpeedreaderTabHelper::OnShowOriginalPage() {
   ReloadContents();
 }
 
+void SpeedreaderTabHelper::ChangeTheme(const std::string& theme) {
+  if (GetSelectedTheme() == theme)
+    return;
+
+  CHECK_EQ(std::u16string::npos, theme.find(u'\''));
+  CHECK_EQ(std::u16string::npos, theme.find(u'\\'));
+
+  constexpr const char16_t kSetTheme[] =
+      uR"js(
+    (function() {
+      const theme = '$1'
+      if (theme == 'system' || theme == '') {
+        document.documentElement.removeAttribute('theme-data')
+      } else {
+        document.documentElement.setAttribute('theme-data', theme)
+      }
+    })();
+  )js";
+
+  const auto script = base::ReplaceStringPlaceholders(
+      kSetTheme, base::UTF8ToUTF16(theme), nullptr);
+
+  web_contents()->GetMainFrame()->ExecuteJavaScriptInIsolatedWorld(
+      script, base::DoNothing(), kIsolatedWorldId);
+  SpeedreaderServiceFactory::GetForProfile(GetProfile())->SelectedTheme(theme);
+}
+
+std::string SpeedreaderTabHelper::GetSelectedTheme() {
+  return SpeedreaderServiceFactory::GetForProfile(GetProfile())
+      ->GetSelectedTheme();
+}
+
+std::string SpeedreaderTabHelper::GetSystemTheme() {
```

Comment:
We can potentially bake this logic to `::GetSelectedTheme` - If the user has no theme selected then it will by default get system's theme. I don't think the consumer of this interface should explicitly handle the fallback.